playwright-tests/tests/matomo-event.spec.ts

```diff
@@ -0,0 +1,170 @@
+        await reefKnotPage.header.connectWalletButton.click();
+        await reefKnotPage.walletListModal.confirmConditionWalletModal();
+        await reefKnotPage.walletListModal.moreWalletsButton.click();
+      });
+
+      await test.step(`Click to wallet ${event.walletName} and check console log`, async () => {
+        const [consoleMessage] = await Promise.all([
+          reefKnotPage.page.waitForEvent('console', (msg) =>
+            msg.text().includes('metrics'),
+          ),
+          reefKnotPage.walletListModal
+            .getWalletInModal(event.walletName)
+            .click(),
+        ]);
+        expect(
+          consoleMessage.text(),
+          `Expected request param "${consoleMessage.text()}" to equal value "${event.eventMessage}"`,
+        ).toContain(event.eventMessage);
+      });
+    });
+  });
+
+  test(
+    qase(431, 'Click to buttons "More wallets" and "Less wallets"'),
+    async () => {
+      const expectedMoreWalletsNameParam = 'more wallets clicked';
+      const expectedLessWalletsNameParam = 'less wallets clicked';
+
+      await test.step('Open the wallet popup', async () => {
+        await reefKnotPage.header.connectWalletButton.click();
+        await reefKnotPage.walletListModal.confirmConditionWalletModal();
+      });
+
+      await test.step('Click to "More wallets"', async () => {
+        const [consoleMessage] = await Promise.all([
+          reefKnotPage.page.waitForEvent('console', (msg) =>
+            msg.text().includes('metrics'),
+          ),
+          reefKnotPage.walletListModal.moreWalletsButton.click(),
+        ]);
+        expect(
+          consoleMessage.text(),
+          `Expected request param "${consoleMessage.text()}" to equal value "${expectedMoreWalletsNameParam}"`,
+        ).toContain(expectedMoreWalletsNameParam);
+      });
+
+      await test.step('Click to "Less wallets"', async () => {
+        const [consoleMessage] = await Promise.all([
+          reefKnotPage.page.waitForEvent('console', (msg) =>
+            msg.text().includes('metrics'),
+          ),
+          reefKnotPage.walletListModal.lessWalletsButton.click(),
+        ]);
+        expect(
+          consoleMessage.text(),
+          `Expected request param "${consoleMessage.text()}" to equal value "${expectedLessWalletsNameParam}"`,
+        ).toContain(expectedLessWalletsNameParam);
+      });
+    },
+  );
+});
+
+test.describe(
+  'ReefKnot. Matomo events [connected wallet]',
+  { tag: [Tags.connectedWallet] },
+  async () => {
+    let reefKnotService: ReefKnotService;
+    let reefKnotPage: ReefKnotPage;
+
+    test.beforeAll(async () => {
+      const browser = await initBrowserWithWallet('metamask');
+      reefKnotService = browser.reefKnotService;
+      reefKnotPage = reefKnotService.reefKnotPage;
+      await reefKnotPage.goto();
+      await reefKnotPage.allowUseCookies();
+    });
+
+    test.beforeEach(async () => {
+      await reefKnotPage.goto();
```

Comment:
why are you opening reefKnot page on each tests and before all suite?

Reply:
fixed after split tests
But usually, I use the `BeforeAll` as preparation steps for a suite, and `BeforeEach` as preparation steps for each test, and sometimes `goto()` can be set up in both places for full control of the environment before tests